R/TealAppDriver.R

```diff
@@ -216,6 +216,39 @@ TealAppDriver <- R6::R6Class( # nolint: object_name.
       self$get_value(output = sprintf("%s-%s", self$active_module_ns(), output_id))
     },
     #' @description
+    #' Get the output from the module's `teal.widgets::table_with_settings` in the `teal` app.
+    #' This function will only access outputs from the name space of the current active teal module.
+    #'
+    #' @param which (integer) If there is more than one table, which should be extracted.
+    #'
+    #' @return The data.frame with table contents.
+    get_active_module_tws_output = function(which = 1) {
+      checkmate::check_number(which, lower = 1)
+      table <-
+        rvest::html_table(
+          self$get_html_rvest(
+            self$active_module_element("table-table-with-settings")
+          ),
+          fill = TRUE
+        )
+      if (identical(table, list())) {
+        table
+      } else {
+        table[[which]]
+      }
+    },
+    #' @description
+    #' Get the output from the module's `teal.widgets::plot_with_settings` in the `teal` app.
+    #' This function will only access plots from the name space of the current active teal module.
+    #'
+    #' @return The `src` attribute as `character(1)` vector.
+    get_active_module_pws_output = function() {
+      self$get_attr(
+        self$active_module_element("myplot-plot_main > img"),
```

Comment:
The same comment applies here, but I don't think `"myplot"` should be an acceptable default value (although it could)

Maybe have no default value and use `app_driver$get_active_module_pws_output("myplot")` every time this method is called

Reply:
@averissimo I've been thinking the same that `myplot-plot_main > img` could be the default value. Looks a bit odd. 

Those are the plot namespace names in `tmc`
<img width="512" alt="image" src="https://github.com/insightsengineering/teal/assets/133694481/d7aef249-baf6-439e-9d7e-30fb8272697a">
and `{tmg}`
<img width="533" alt="image" src="https://github.com/insightsengineering/teal/assets/133694481/959cc8a4-baa1-426d-870c-99332848253a">

Reply:
For sure hardcoded `myplot` will not work for all modules. We can allow to pass plot namespace value as a parameter, and we could rewrite the function to:

```r
    get_active_module_pws_output = function(plot_name) {
      self$get_attr(
        self$active_module_element(sprintf("%s-plot_main > img", plot_name)),
```

naming the parameter `plot_name` or `plot_namespace` or `plot_ns`

Reply:
That's the exact code I was writing  :-D to suggest :+1: